app/src/main/java/io/gnosis/safe/ui/safe/add/AddSafeViewModel.kt

```diff
@@ -18,17 +18,16 @@ class AddSafeViewModel
         safeLaunch {
             runCatching {
                 updateState { AddSafeState(ViewAction.Loading(true)) }
-                val validSafe = safeRepository.isValidSafe(address.asEthereumAddress() ?: throw InvalidSafeAddress())
+                val newAddress = address.asEthereumAddress() ?: throw InvalidSafeAddress
+                val validSafe = safeRepository.isValidSafe(newAddress)
+                val isAddressUsed = safeRepository.isSafeAddressUsed(newAddress)
```

Comment:
I would reverse this check sequence. If safe is used it will be valid, so no need to check the validity afterward. Otherwise, we need also check validity.

Reply:
Good call 👍  I wrote it in a way that we make the network call if it's absolutely necessary